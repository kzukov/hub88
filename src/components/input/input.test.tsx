import React from "react";
import Input from "./input";
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("<Input />", () => {
  test("matches snapshot", () => {
    const { asFragment } = render(<Input />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("fires onChange event if input changes", async () => {
    const testvalue = "testvalue";

    render(<Input />);

    const input = screen.getByRole<HTMLInputElement>("textbox");

    fireEvent.change(input, { target: { value: testvalue } });

    expect(input.value).toBe(testvalue);
  });
});
