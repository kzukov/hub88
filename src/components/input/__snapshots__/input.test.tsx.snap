// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> matches snapshot 1`] = `
<DocumentFragment>
  <input
    class="form-control"
    type="text"
  />
</DocumentFragment>
`;
